presto-main/src/main/java/com/facebook/presto/testing/LocalQueryRunner.java

```diff
@@ -1056,28 +1059,25 @@ public Plan createPlan(Session session, @Language("SQL") String sql, List<PlanOp
                 costCalculator,
                 dataDefinitionTask,
                 distributedPlanChecker);
-        Analyzer analyzer = new Analyzer(session, metadata, sqlParser, accessControl, Optional.of(queryExplainer), preparedQuery.getParameters(), parameterExtractor(preparedQuery.getStatement(), preparedQuery.getParameters()), warningCollector);
 
-        Analysis analysis = analyzer.analyze(preparedQuery.getStatement());
+        BuiltInQueryAnalyzer queryAnalyzer = new BuiltInQueryAnalyzer(metadata, sqlParser, accessControl, Optional.of(queryExplainer));
 
-        final PlanVariableAllocator planVariableAllocator = new PlanVariableAllocator();
-        LogicalPlanner logicalPlanner = new LogicalPlanner(
-                session,
-                idAllocator,
-                metadata,
-                planVariableAllocator);
+        AnalyzerContext analyzerContext = getAnalyzerContext(queryAnalyzer, idAllocator, new PlanVariableAllocator(), session);
+
+        QueryAnalysis queryAnalysis = queryAnalyzer.analyze(analyzerContext, preparedQuery);
+        queryAnalyzer.checkAccessPermissions(analyzerContext, queryAnalysis);
 
         PlanNode planNode = session.getRuntimeStats().profileNanos(
                 LOGICAL_PLANNER_TIME_NANOS,
-                () -> logicalPlanner.plan(analysis));
+                () -> queryAnalyzer.plan(analyzerContext, queryAnalysis));
 
         Optimizer optimizer = new Optimizer(
                 session,
                 metadata,
                 optimizers,
                 singleNodePlanChecker,
                 sqlParser,
-                planVariableAllocator,
+                (PlanVariableAllocator) analyzerContext.getVariableAllocator(),
```

Comment:
is it safe to do this cast?

Reply:
Yes, our plan nodes only work with PlanVariableAllocator. I can add a cast check in the Optimizer if you want.

Reply:
AnalyzerContext only returns a VariableAllocator though.  i don't like that we assume it's always a planVariableAllocator.

Reply:
> AnalyzerContext only returns a VariableAllocator though. i don't like that we assume it's always a planVariableAllocator.

Yeah I got this issue in my PR as well. Not sure why we need two concepts for things that are just for convenience. Unnecessary complexity.

Reply:
Will add a cast.

Reply:
Actually I have changed the constructor for Optimizer to take the VariableAllocator, and added a cast there.